Allow Sanitization of Non String Params

Bene-Graham opened the pull request:
This should act like how the validator part of the library works, by toString() the value then perform the action against it

This should deal with my issue #580 

The one thing I was not sure on, is where to put the tests that the value values are passed and sanitized correctly. I ended up creating a sanitizer.spec.js in the check folder.

Bene-Graham commented:
I fixed the linting issues

gustavohenke commented:
Feel free to reply with your thoughts after my review above.
There's no need to start changing the code straight away 🙂

Bene-Graham commented on utils/select-fields.js:
```diff
@@ -70,16 +71,14 @@ function expand(object, path, paths) {
-      callSanitizer(sanitizer, prev) :
-      prev.value;
-
+    const value = !!prev.value ? callSanitizer(sanitizer, prev) : prev.value;
```
Not sure why I needed to a this check. If ended up just calling callSanitizer(sanitizer, prev) test would fail. Mainly because the prev.value is undefined.

Bene-Graham commented:
> I'm not sure if always converting the value back to string is the desired behaviour.
validator.js is unable to deal with non-string values, so that's fine for standard sanitizers, but maybe not for custom ones.

You are 100% right. Miss read the validator code in the runner.js file.

>Also, I'm not sure if the tests with the standard validators (toBoolean, toInt) are relevant; most of the behaviour is guaranteed by other means.

OK, I have removed the tests
---

I also added a comment to a line of code, that I needed to add to make the test pass.

alzeih commented:
I’m not sure converting to a string is the right approach for object and array like types?

If a new `customSanitizer` method (with a new name) was created that didn’t use the validator.js library then it could call the custom sanitizer code with whatever input, not just if the value is a string, and wouldn’t break existing code. It might mean that this new custom sanitizer couldn’t be chained with other sanitizers from validator.js though, or only can be chained effectively if it returns a string to other sanitizers.

Bene-Graham commented:
> I’m not sure converting to a string is the right approach for object and array like types?
> 
> If a new `customSanitizer` method (with a new name) was created that didn’t use the validator.js library then it could call the custom sanitizer code with whatever input, not just if the value is a string, and wouldn’t break existing code. It might mean that this new custom sanitizer couldn’t be chained with other sanitizers from validator.js though, or only can be chained effectively if it returns a string to other sanitizers. #632 addresses this.

If I am reading #632 correctly it won't solve the issue where the built in sanitizers won't run when the values are not already strings.

alzeih commented:
Yes, the second half of that comment is not relevant here (and is deleted).

String conversion for object and array like types though sounds like it could have some problems though to me?

gustavohenke commented:
Hello! A fix should has been published in [`v6.0.0`](https://github.com/express-validator/express-validator/releases/tag/v6.0.0) 🚢, please try it out!
